DashAI/back/api/api_v1/endpoints/plugins.py

```diff
@@ -0,0 +1,284 @@
+import logging
+from typing import Callable, List
+
+from dependency_injector.wiring import Provide, inject
+from fastapi import APIRouter, Depends, Response, status
+from fastapi.exceptions import HTTPException
+from sqlalchemy import exc
+from sqlalchemy.orm import Session
+from typing_extensions import ContextManager
+
+from DashAI.back.api.api_v1.schemas.plugin_params import (
+    PluginParams,
+    PluginUpdateParams,
+)
+from DashAI.back.containers import Container
+from DashAI.back.dependencies.database.models import Plugin, Tag
+from DashAI.back.plugins.utils import get_plugins_from_pypi
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+router = APIRouter()
+
+
+@router.get("/")
+@inject
+async def get_plugins(
+    session_factory: Callable[..., ContextManager[Session]] = Provide[
+        Container.db.provided.session
+    ],
+):
+    """Retrieve a list of the stored plugins in the database.
+
+    Parameters
+    ----------
+    session_factory : Callable[..., ContextManager[Session]]
+        A factory that creates a context manager that handles a SQLAlchemy session.
+        The generated session can be used to access and query the database.
+
+    Returns
+    -------
+    List[dict]
+        A list of dictionaries representing the found plugins.
+        Each dictionary contains information about the plugin, including its name,
+        author, tags, description and creation date.
+        If no plugins are found, an empty list will be returned.
+    """
+    with session_factory() as db:
+        try:
+            plugins = db.query(Plugin).all()
+
+        except exc.SQLAlchemyError as e:
+            logger.exception(e)
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail="Internal database error",
+            ) from e
+
+    return plugins
+
+
+@router.get("/{plugin_id}")
+@inject
+async def get_plugin(
```

Comment:
Since we have tags, I think it would be prudent to also have the option to bring some plugins according to their tag or a set of tags ( by making a union of sets).

Reply:
Also, wouldn't it be nice to be able to filter by status?
This could make it much easier to get the plugins that are installed in a view, eventually designed to manage them.